pkg/base/cockpit.js

```diff
@@ -2510,6 +2510,62 @@ function full_scope(cockpit, $) {
         return new HttpClient(endpoint, options || { });
     };
 
+    /* ---------------------------------------------------------------------
+     * Permission
+     */
+
+    var authority = null;
+
+    function Permission(options) {
+        var self = this;
+        self.allowed = null;
+
+        var user = cockpit.user;
+        var group = null;
+
+        if (options)
+            group = options.group;
+
+        function decide() {
+            var allowed = null;
+
+            if (user.id === 0)
+                allowed = true;
+            else if (!group)
+                allowed = false;
+
+            if (!allowed && group && user.groups) {
+                allowed = false;
+                $.each(user.groups, function(i, name) {
+                    if (name == group) {
+                        allowed = true;
+                        return false;
```

Comment:
When you ```return false``` from an each callback, the iteration is cancelled. http://api.jquery.com/jquery.each/